src/ziggurat/kafka_consumer/consumer.clj

```diff
@@ -31,7 +33,8 @@
               ConsumerConfig/ENABLE_AUTO_COMMIT_CONFIG       enable-auto-commit
               ConsumerConfig/AUTO_COMMIT_INTERVAL_MS_CONFIG  (int commit-interval-ms)
               ConsumerConfig/KEY_DESERIALIZER_CLASS_CONFIG   key-deserializer-class-config
-              ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG value-deserializer-class-config})))
+              ConsumerConfig/VALUE_DESERIALIZER_CLASS_CONFIG value-deserializer-class-config
+              ConsumerConfig/DEFAULT_API_TIMEOUT_MS_CONFIG   (int default-api-timeout-ms)})))
```

Comment:
Can't `default-api-timeout-ms` take care of type casting?

Reply:
values coming from the config has a type of Long, ConsumerConfig/DEFAULT_API_TIMEOUT_MS_CONFIG expects a 32 bit integer hence the type cast...